code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java

```diff
@@ -50,17 +50,98 @@ public boolean includes(int value) {
 
     //function that returns a collection of all the values in our linked list
     public int[] print() {
-        if (this.head == null) {
+        try {
+            if (this.head == null) {
+                return null;
+            }
+            int[] resultArr = new int[size];
+            Node current = head;
+            int arrCnt = 0;
+            while (current != null) {
+                resultArr[arrCnt] = current.value;
+                current = current.next;
+                arrCnt++;
+            }
+            return resultArr;
+
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("The index your are trying to access is out of bounds");
             return null;
         }
-        int[] resultArr = new int[size];
-        Node current = head;
-        int arrCnt = 0;
-        while (current != null) {
-            resultArr[arrCnt] = current.value;
-            current = current.next;
-            arrCnt++;
+    }
+
+
+    //This function adds a new value to the end of the linked list
+    public void append(int value){
+        Node current = new Node();
```

Comment:
You could also make your Node constructor take in an argument that it sets as the value